elasticsearch-singledocument/src/main/java/org/vertexium/elasticsearch/ElasticsearchSingleDocumentSearchIndex.java

```diff
@@ -718,16 +880,37 @@ private String getVisibilityHash(Graph graph, String propertyName, Visibility vi
 
     @Override
     public void deleteElement(Graph graph, Element element, Authorizations authorizations) {
+        deleteExtendedDataForElement(element);
+
         String indexName = getIndexName(element);
         String id = element.getId();
         if (MUTATION_LOGGER.isTraceEnabled()) {
             LOGGER.trace("deleting document %s", id);
         }
-        getClient().delete(
-                getClient()
-                        .prepareDelete(indexName, ELEMENT_TYPE, id)
-                        .request()
-        ).actionGet();
+        getClient().prepareDelete(indexName, ELEMENT_TYPE, id).execute().actionGet();
+    }
+
+    private void deleteExtendedDataForElement(Element element) {
+        try {
+            FilterBuilder filters[] = new FilterBuilder[1];
+            filters[0] = FilterBuilders.termFilter(ElasticsearchSingleDocumentSearchIndex.EXTENDED_DATA_ELEMENT_ID_FIELD_NAME, element.getId());
+            FilterBuilder filter = FilterBuilders.andFilter(filters);
```

Comment:
Can we add a andFilter method that takes a single filter and calls the addFilter(filters[])?